src/http/structs.rs

```diff
@@ -52,7 +49,10 @@ impl Database {
         Database {
             entity: entity.slug.clone(),
             database: database.slug.clone(),
-            database_type: DBType::from_i16(database.db_type).to_str().to_string(),
+            database_type: DBType::try_from(database.db_type)
+                .expect("unknown database type")
```

Comment:
The nice thing about the pre-macro model is that the error strings were generated in one location, and you could even include the values in the panic. Would it be possible to push the error strings/values into the macros?

Reply:
Fixed with 2b20d875ecb1e8f5f6cceca021c82a4dff63e9d5 :)